src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +61,646 @@ const ActorInit En_Rz_InitVars = {
+        return true;
+    }
+    return false;
+}
+
+s32 EnRz_UpdateSkelAnime(EnRz* this, PlayState* play) {
+    if (this->animIndex < EN_RZ_ANIM_LINK_NORMAL_WAIT_FREE) {
+        return SkelAnime_Update(&this->skelAnime);
+    } else {
+        return LinkAnimation_Update(play, &this->skelAnime);
+    }
+}
+
+EnRz* EnRz_FindSister(EnRz* this, PlayState* play) {
+    Actor* npc = play->actorCtx.actorLists[ACTORCAT_NPC].first;
+
+    while (npc != NULL) {
+        if ((npc->id == ACTOR_EN_RZ) && (EN_RZ_GET_TYPE(&this->actor) == EN_RZ_GET_TYPE(npc))) {
+            if (&this->actor != npc) {
+                return (EnRz*)npc;
+            }
+        }
+        npc = npc->next;
+    }
+    return NULL;
+}
+
+void func_80BFBDFC(PlayState* play) {
+    if (gSaveContext.save.weekEventReg[75] & 0x80) {
+        func_80151BB4(play, 0x27);
+    }
+    func_80151BB4(play, 0xC);
+}
+
+void EnRz_Destroy(Actor* thisx, PlayState* play) {
+    EnRz* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+s32 func_80BFBE70(EnRz* this, PlayState* play) {
+    u16 action;
+
+    if ((EN_RZ_GET_SISTER(&this->actor) == EN_RZ_JUDO) && (this->animIndex == EN_RZ_ANIM_APPLAUDING)) {
+        func_800B9010(&this->actor, NA_SE_EV_CLAPPING_2P - SFX_FLAG);
+    }
+
+    if (Cutscene_CheckActorAction(play, this->csAction)) {
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, this->csAction));
+        action = play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->csAction)]->action;
+        if (this->actionIndex != action) {
+            this->actionIndex = action;
+            switch (action & 0xFFFF) {
+                case 1:
+                    func_80BFBA1C(play, this, EN_RZ_ANIM_STANDING);
+                    break;
+
+                case 2:
+                    func_80BFBA1C(play, this, EN_RZ_ANIM_ON_KNEES);
+                    break;
+
+                case 3:
+                    func_80BFBA1C(play, this, EN_RZ_ANIM_APPLAUDING);
+                    break;
+
+                case 4:
+                    func_80BFBA1C(play, this, EN_RZ_ANIM_LINK_DANCE);
+                    break;
+            }
+        }
+        return true;
+    }
+    return false;
+}
+
+s32 func_80BFBFAC(EnRz* this, PlayState* play) {
+    if (this->actor.cutscene == -1) {
+        Message_StartTextbox(play, 0x2925, &this->actor);
+        this->actionFunc = func_80BFC078;
+    } else if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        return false;
+    } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_Start(this->actor.cutscene, &this->actor);
+        return true;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+    return false;
+}
+
+void EnRz_SetupUpdateSkelAnime(EnRz* this, PlayState* play) {
```

Comment:
Hmm, it doesn't really seem like this is setting up `UpdateSkelAnime`. Instead it is just an action function whose only job is to `UpdateSkelAnime`. Not sure what to name it, and probably depends on all the other action functions. Personally I think it would be best to just leave it unnamed for the time being.